config/v1/types_infrastructure.go

```diff
@@ -562,6 +617,37 @@ type BareMetalPlatformStatus struct {
 	// datacenter DNS, a DNS service is hosted as a static pod to serve those hostnames
 	// to the nodes in the cluster.
 	NodeDNSIP string `json:"nodeDNSIP,omitempty"`
+
+	// loadBalancer defines how the load balancer used by the cluster is configured.
+	// +default={"openshiftManaged": {"type": "Default"}, "type": "OpenShiftManaged"}
+	// +kubebuilder:default={"openshiftManaged": {"type": "Default"}, "type": "OpenShiftManaged"}
```

Comment:
Neat, I didn't realise you could do that!

I wonder if we want to do this specifically, though. Default could probably just remain empty?

Reply:
I'll defer the answer to @JoelSpeed / @deads2k for this one. I thought we wanted to have a (safe) default so it'll be in `PlatformStatus` no matter if the user feed this fields or not.

Reply:
Defaulting it means that we can then validate that the field is never changed, since we don't expect the default implementation to ever change, and we don't want to support transitions yet, this seems like a benefit there, since we know all existing clusters will be using the default implementation, we can then validate transitions later